full/src/main/java/apoc/mongodb/Mongo.java

```diff
@@ -0,0 +1,114 @@
+package apoc.mongodb;
+
+import apoc.result.LongResult;
+import apoc.result.MapResult;
+import org.neo4j.logging.Log;
+import org.neo4j.procedure.Context;
+import org.neo4j.procedure.Description;
+import org.neo4j.procedure.Name;
+import org.neo4j.procedure.Procedure;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static apoc.mongodb.MongoDBUtils.Coll;
+import static apoc.mongodb.MongoDBUtils.getDocument;
+import static apoc.mongodb.MongoDBUtils.getMongoColl;
+
+public class Mongo {
+
+    @Context
+    public Log log;
+
+    @Procedure("apoc.mongo.aggregate")
+    @Description("apoc.mongo.aggregate(uri, pipeline, $config) yield value - perform an aggregate operation on mongodb collection")
+    public Stream<MapResult> aggregate(@Name("uri") String uri, @Name("pipeline") List<Map<String, Object>> pipeline, @Name(value = "config", defaultValue = "{}") Map<String, Object> config) {
+        MongoDbConfig conf = new MongoDbConfig(config);
+        return executeMongoQuery(uri, conf, 
+                coll -> coll.aggregate(pipeline.stream().map(MongoDBUtils::getDocument).collect(Collectors.toList())).map(MapResult::new), 
```

Comment:
`pipeline.stream().map(MongoDBUtils::getDocument).collect(Collectors.toList())` me lo metti in una variabile a parte?